haddock-api/src/Haddock/Interface.hs

```diff
@@ -165,6 +172,9 @@ createIfaces verbosity flags instIfaceMap mods = do
 processModule :: Verbosity -> ModSummary -> [Flag] -> IfaceMap -> InstIfaceMap -> Ghc (Maybe Interface)
 processModule verbosity modsum flags modMap instIfaceMap = do
   out verbosity verbose $ "Checking module " ++ moduleString (ms_mod modsum) ++ "..."
+#if defined(mingw32_HOST_OS)
```

Comment:
Please put it in `haddockWithGhc` it is the toplevel entrypoint. So we are sure nothing slips through.